homeassistant/components/escea/climate.py

```diff
@@ -0,0 +1,217 @@
+                return
+            self.set_available(True)
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_RECONNECTED, controller_reconnected
+            )
+        )
+
+        @callback
+        def controller_update(ctrl: Controller) -> None:
+            """Handle controller data updates."""
+            if ctrl is not self._controller:
+                return
+            self.async_write_ha_state()
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_UPDATE, controller_update
+            )
+        )
+
+    @callback
+    def set_available(self, available: bool, ex: Exception = None) -> None:
+        """Set availability for the controller."""
+        if self._attr_available == available:
+            return
+
+        if available:
+            _LOGGER.info("Reconnected controller %s ", self._controller.device_uid)
+        else:
+            _LOGGER.info(
+                "Controller %s disconnected due to exception: %s",
+                self._controller.device_uid,
+                ex,
+            )
+
+        self._attr_available = available
+        self.async_write_ha_state()
+
+    @property
+    def device_info(self):
+        """Return the device info for the Escea system."""
+        return {
+            "identifiers": {(ESCEA, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Escea",
+        }
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return current operation ie. heat, cool, idle."""
+        return HVAC_MODE_HEAT if self._controller.is_on else HVAC_MODE_OFF
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        return self._controller.current_temp
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        return self._controller.desired_temp
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the fan setting."""
+        return _ESCEA_FAN_TO_HA[self._controller.fan]
+
+    async def wrap_and_catch(self, coro: Coroutine):
+        """Catch any connection errors and set unavailable."""
+        try:
+            await coro
+        except ConnectionError as ex:
+            self.set_available(False, ex)
+        else:
+            self.set_available(True)
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp is not None:
+            await self.wrap_and_catch(self._controller.set_desired_temp(temp))
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set new target fan mode."""
+        await self.wrap_and_catch(self._controller.set_fan(_HA_FAN_TO_ESCEA[fan_mode]))
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
```

Comment:
`hvac_mode` is an enum nowadays.

https://developers.home-assistant.io/docs/core/entity/climate#properties

Reply:
```suggestion
    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
```